app/src/main/java/com/project/app/base/BaseFragment.kt

```diff
@@ -0,0 +1,158 @@
+package com.project.app.base
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.os.Bundle
+import android.view.*
+import android.view.inputmethod.InputMethodManager
+import android.widget.ImageButton
+import android.widget.TextView
+import android.widget.Toast
+import androidx.constraintlayout.widget.ConstraintLayout
+import androidx.core.content.ContextCompat
+import androidx.fragment.app.Fragment
+import androidx.navigation.fragment.findNavController
+import androidx.viewbinding.ViewBinding
+import com.project.app.R
+import com.project.app.helpers.LoadingDialog
+import com.project.app.helpers.ToolbarType
+import kotlinx.coroutines.Job
+
+abstract class BaseFragment<VB : ViewBinding> : Fragment(), IBaseFragment {
+    lateinit var activityIndicator: LoadingDialog
+    private var job = Job()
+    private lateinit var topBarLayout: ConstraintLayout
+    lateinit var leftButtonToolbar: ImageButton
+    private lateinit var textViewMenuTitle: TextView
+    private lateinit var rigthButtonToolbar: ImageButton
+
+    private lateinit var fragmentView: View
+
+    private var _binding: ViewBinding? = null
+    abstract val bindingInflater: (LayoutInflater, ViewGroup?, Boolean) -> VB
+    @Suppress("UNCHECKED_CAST")
+    val binding: VB
+        get() = _binding as VB
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        _binding = bindingInflater.invoke(inflater, container, false)
+        return requireNotNull(_binding).root
+    }
+
+    override fun onDestroyView() {
+        super.onDestroyView()
+        _binding = null
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        topBarLayout = requireActivity().findViewById(R.id.topBarLayout)
+        leftButtonToolbar = requireActivity().findViewById(R.id.leftButtonToolbar)
+        rigthButtonToolbar = requireActivity().findViewById(R.id.rigthButtonToolbar)
+        textViewMenuTitle = requireActivity().findViewById(R.id.menuItemTitle)
+        fragmentView = view
+        activityIndicator = LoadingDialog(requireActivity())
```

Comment:
Lo mismo acá, podríamos usar el activity que nos guardamos en una variable quizás.

```suggestion
        activityIndicator = LoadingDialog(activity)
```